Flaky test: TestProgress/single_step_delayed_failure

## Flaky Test

* **Test Name:** TestProgress/single_step_delayed_failure
* **Link:** https://github.com/elastic/elastic-agent/blob/123ba9ce80c9865f72fa3659b5cafe9b51954f49/internal/pkg/agent/install/progress_test.go#L45-L60
* **Branch:** https://fleet-ci.elastic.co/job/elastic-agent/job/elastic-agent-mbp/job/PR-3482/
* **Artifact Link:** https://fleet-ci.elastic.co/job/elastic-agent/job/elastic-agent-mbp/job/PR-3482/7/artifact/
* **Notes:** It seems the regexp matching is flaky (mostly on windows). The number of dots in the log seems to be off by 1

### Stack Trace

```
=== RUN   TestProgress/single_step_delayed_failure
    progress_test.go:59: 
        	Error Trace:	C:/Users/jenkins/workspace/_agent_elastic-agent-mbp_PR-3482/src/github.com/elastic/elastic-agent/internal/pkg/agent/install/progress_test.go:59
        	Error:      	Expect "step 1 starting... FAILED
        	            	" to match "step 1 starting\.{3}\.+ FAILED\n"
        	Test:       	TestProgress/single_step_delayed_failure
--- FAIL: TestProgress/single_step_delayed_failure (0.02s)
```

--- a/internal/pkg/agent/install/progress_test.go
+++ b/internal/pkg/agent/install/progress_test.go
@@ -51,7 +51,7 @@ func TestProgress(t *testing.T) {
 		rs := pt.Start()
 
 		s := rs.StepStart("step 1 starting")
-		time.Sleep(15 * time.Millisecond) // to simulate work being done
+		time.Sleep(100 * time.Millisecond) // to simulate work being done
 		s.Failed()
 
 		rs.Failed()

